File: C++/NucleusLib/pnSceneObject/plAudioInterface.cpp
#include "plAudioInterface.h"
#include "../../PubUtilLib/plResMgr/plResManager.h"

plAudioInterface::plAudioInterface(PlasmaVer pv) {
    Audible = new plKey();
}

plAudioInterface::~plAudioInterface() {
    Audible->UnRef();
}

void plAudioInterface::read(hsStream *S) {
    plObjInterface::read(S);
    Audible = plResManager::inst->readKey(S);
    Audible->Ref();
}

void plAudioInterface::write(hsStream *S) {
    plObjInterface::write(S);
    plResManager::inst->writeKey(S, Audible);
}

File: Java/Plasma/NucleusLib/pnSceneObject/plAudioInterface.java
package Plasma.NucleusLib.pnSceneObject;

import java.io.IOException;
import Plasma.CoreLib.hsStream;
import Plasma.NucleusLib.pnKeyedObject.*;

public class plAudioInterface extends plObjInterface {
    protected plKey Audible = new plKey();

    public void read(hsStream S) throws IOException {
        super.read(S);
        Audible.readRef(S);
    }

    public void write(hsStream S) throws IOException {
        super.write(S);
        Audible.writeRef(S);
    }
}

